src/app/widgets/map/map.component.ts

```diff
@@ -61,39 +74,40 @@ export class MapComponent implements OnInit, OnDestroy {
     this.map.addCatalog(cat);
 
 
-    var n_sources = Object.keys(catalog.Source_Name).length;
+    var n_sources = catalog.data.index.length;
     console.log(catalogName, " # of sources: ", n_sources);
 
     for (var i = 0; i < n_sources; i++) {
 
-      var row = i.toString();
+      // var row = i.toString();
 
-      var round_ra = (Math.round(catalog['RAJ2000'][row] * 100) / 100).toFixed(2);
-      var round_dec = (Math.round(catalog['DEJ2000'][row] * 100) / 100).toFixed(2);
-      var round_glon = (Math.round(catalog['GLON'][row] * 100) / 100).toFixed(2);
-      var round_glat = (Math.round(catalog['GLAT'][row] * 100) / 100).toFixed(2);
+      var round_ra = (Math.round(catalog.getSourceByRowIndex(i).data[raType] * 100) / 100).toFixed(2);
+      var round_dec = (Math.round(catalog.getSourceByRowIndex(i).data[decType] * 100) / 100).toFixed(2);
+      var round_glon = (Math.round(catalog.getSourceByRowIndex(i).data.GLON * 100) / 100).toFixed(2);
+      var round_glat = (Math.round(catalog.getSourceByRowIndex(i).data.GLAT * 100) / 100).toFixed(2);
```

Comment:
For this, maybe introduce a convenience method on the base catalog class that takes the column name and number of digits. Maybe call it `get_val`?